### PR TITLE
chore: misc code cleanups

### PR DESCRIPTION
**Pull request by tearfur**
No functional changes intended. Mainly readability improvements.

### CODE REVIEW

**Review by Coeur — APPROVED** [review]

File: libtransmission/global-ip-cache.cc
Comment (mikedld) [inline libtransmission/global-ip-cache.cc:87]: This feels too close to the good old "assignment within a conditional expression" type of warning, not sure I like it.

File: libtransmission/global-ip-cache.cc
Comment (mikedld) [inline libtransmission/global-ip-cache.cc:151]: Why not `std::make_unique<>()`?

Comment (tearfur) [reply libtransmission/global-ip-cache.cc:151]: Because the constructor is private.

File: libtransmission/global-ip-cache.cc
Comment (tearfur) [reply libtransmission/global-ip-cache.cc]: c1906837b050d76af4311e6b6d3d0358946c0981

File: libtransmission/global-ip-cache.cc
Comment (mikedld) [reply libtransmission/global-ip-cache.cc]: I think the usual way to handle this is to make the ctor public, but with an additional parameter of private type.

OTOH I don't see any secret in the way this object is contructed, the `create()` method parameters match ctor parameters exactly, so why make the ctor private and need this method in the first place?..

Comment (tearfur) [reply libtransmission/global-ip-cache.cc]: I think I was just trying to keep the API consistent with the other existing classes, because a lot of them had `create()` too, but you're right that this class doesn't need it.

I'll make the refactor later.

Comment (tearfur) [reply libtransmission/global-ip-cache.cc]: 760c812c805040c51161cbe71881838889f245fc

File: libtransmission/CMakeLists.txt
Comment (Coeur) [inline libtransmission/CMakeLists.txt:68]: renamed files need a change in xcodeproj, so removing my approval for now

**Review by Coeur — CHANGES_REQUESTED** [review]

File: libtransmission/CMakeLists.txt
Comment (Coeur) [reply libtransmission/CMakeLists.txt:68]: Attaching my xcodeproj patch:
[6927.patch](https://github.com/user-attachments/files/16183741/6927.patch)

Comment (tearfur) [reply libtransmission/CMakeLists.txt:68]: dbf193ee0e77a8c6e31d5c13c45d32f4267f608b Thanks.

**Review by Coeur — APPROVED** [review]

**Comment by mikedld** [comment]
As you can see TeamCity was green before the most recent fix but only because I've disabled Mac (including Xcode) builds due to a connectivity issue with the MacStadium-provided builder. I'll wait for it to be resolved before merging. If someone wants to contribute xcodebuild-based GitHub Actions configuration, that'd be appreciated.

**Comment by tearfur** [comment]
@nevack This step has no effect because Xcode is not used in GitHub CI.

https://github.com/transmission/transmission/blob/7f79cb16ee194d58ce665f9319524bc5e6e4f91d/.github/workflows/actions.yml#L352-L354

**Comment by nevack** [comment]
> Xcode is not used in GitHub CI.

WDYM?

https://github.com/transmission/transmission/actions/runs/9900791761/job/27352160799?pr=6927#step:6:20

https://github.com/transmission/transmission/actions/runs/9900791761/job/27352160799?pr=6927#step:6:28

https://github.com/transmission/transmission/actions/runs/9900791761/job/27352160799?pr=6927#step:6:35

`xcodebuild` itself is not used in this particular actions, but sysroot/compilers/sdks are used correctly

**Comment by tearfur** [comment]
Ooh, my bad. I'm really not familiar with the Mac side of things.

**Review by mikedld — APPROVED** [review]

**Comment by Coeur** [comment]
> If someone wants to contribute xcodebuild-based GitHub Actions configuration, that'd be appreciated.

3rd commit of #6911

**Pull request merged**
